src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs

```diff
@@ -25,33 +24,32 @@ public class BlittableObjectInstance : ObjectInstance
         public readonly string ChangeVector;
         public readonly BlittableJsonReaderObject Blittable;
         public readonly string DocumentId;
-        public HashSet<string> Deletes;
-        public Dictionary<string, BlittableObjectProperty> OwnValues = 
-            new Dictionary<string, BlittableObjectProperty>();
+        public HashSet<Key> Deletes;
+        public Dictionary<Key, BlittableObjectProperty> OwnValues = new Dictionary<Key, BlittableObjectProperty>();
```

Comment:
As you anyway have custom build of Jint, there's also possibility to make StringDictionarySlim public and use it. It's basically the [tuned version of Dictionary](https://github.com/dotnet/corefxlab/blob/master/src/Microsoft.Experimental.Collections/Microsoft/Collections/Extensions/DictionarySlim.cs) ([some history here](https://github.com/dotnet/corefxlab/pull/2458)).

Reply:
Or maybe it is better to include 'Microsoft.Experimental.Collections' and use `DictionarySlim<TK, TV>` instead of Jint's string version. What are benefits of `StringDictionarySlim<T>` vs `DictionarySlim<string, T>` or `DictionarySlim<Jint.Key, T>` ?

Last one seems like the most reasonable option, because we would have an opportunity to take leverage from `Key`. Thoughts?

Reply:
Probably the most convenient path is to use the MS Experimental. Jint version just skips GetHashCode etc and accesses the struct field directly.